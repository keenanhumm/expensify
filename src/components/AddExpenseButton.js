import React from 'react';
import { NavLink } from 'react-router-dom';

export default () => (
    <div className="container">
        <NavLink to="/create" className="nav-link" activeStyle={{
            color: 'white',
            fontWeight: 'bold',
            textDecoration: 'none'
        }}>
            <img src="images/add.gif" height="40px" width="40px" /> New Expense
        </NavLink>
    </div>
);