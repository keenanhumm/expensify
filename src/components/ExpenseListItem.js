import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';


export const ExpenseListItem = ({id, description, amount, createdAt}) => (
    <div>
        <Link to={`/edit/${id}`}> 
            <h3>{description}</h3>
        </Link>
        <p>Amount: ${amount/ 100}</p>
        <p>Created on {moment(createdAt).format('MMM DD YYYY')} at {moment(createdAt).format('hh:mma')}</p>
    </div>
);  

export default ExpenseListItem;