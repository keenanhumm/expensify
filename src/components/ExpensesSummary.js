import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
    const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedAmount = numeral(expensesTotal / 100).format('$0, 0.00');

    return (
        <div>
            <h3>
                Viewing {expensesCount} {expenseWord}, with a grand total of {formattedAmount}.
            </h3>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: expensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);