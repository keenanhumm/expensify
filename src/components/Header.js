import React from 'react';
import { NavLink } from 'react-router-dom';
import { Image, Button } from 'react-bootstrap';

const Header = () => {
    return (
        <header>
            <div className="container">
                <NavLink to="/" className="nav-link" exact={true}><Image src="images/biglogo.png" responsive /></NavLink>
            </div>
        </header>
    );

};

export default Header;