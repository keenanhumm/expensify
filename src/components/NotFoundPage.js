import React from 'react';
import { Link } from 'react-router-dom';
export const NotFoundPage = () => (
    <div>
        <h1>404! Page Not Found</h1>
        <h2><Link to="/">Go Home</Link></h2>
    </div>
);

export default NotFoundPage;