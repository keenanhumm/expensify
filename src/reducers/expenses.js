
const expenseReducerDefaultState = [];
export default (state = expenseReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [
                ...state,
                action.expense
            ];
        case 'REMOVE_EXPENSE':
            return state.filter(({ id }) => (id !== action.id));
        case 'EDIT_EXPENSE':
            return state.map((expense) => {
                if (expense.id === action.id) {
                    //override existing expense info with whatever updates were passed in
                    return {
                        ...expense,
                        ...action.updates
                    };
                } else {
                    //just return the unchanged expense 
                    return expense;
                }
            });
        default:
            return state;
    }
};  