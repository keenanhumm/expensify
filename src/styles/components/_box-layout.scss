.box-layout {
    background: url('/images/city-light.jpg');
    height: 10vh;
    height: 100vh;
}