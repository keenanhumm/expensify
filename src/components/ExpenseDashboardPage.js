import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';
import AddExpenseButton from './AddExpenseButton';

export const ExpenseDashboardPage = () => (
    <div>
        <AddExpenseButton />
        <ExpenseListFilters />
        <div className="box-layout">
            <ExpensesSummary />
            <ExpenseList /> 
        </div>
       
    </div>
);


export default ExpenseDashboardPage;