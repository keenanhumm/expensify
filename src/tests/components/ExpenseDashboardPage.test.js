import { shallow } from 'enzyme';
import React from 'react';
import { ExpenseDashboardPage } from '../../components/ExpenseDashboardPage';
import expenses from '../fixtures/expenses';

test('should render expense dashboard correctly', () => {
    const wrapper = shallow(<ExpenseDashboardPage />);
    expect(wrapper).toMatchSnapshot();
});