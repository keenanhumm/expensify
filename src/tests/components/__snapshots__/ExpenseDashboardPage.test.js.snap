// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expense dashboard correctly 1`] = `
<div>
  <Component />
  <Connect(ExpenseListFilters) />
  <div
    className="box-layout"
  >
    <Connect(ExpensesSummary) />
    <Connect(ExpenseList) />
  </div>
</div>
`;
